sql/plsql/PlSqlLexer.g4

```diff
@@ -2379,10 +2380,13 @@ PERIOD:         '.';
 UNSIGNED_INTEGER:    [0-9]+;
 APPROXIMATE_NUM_LIT: FLOAT_FRAGMENT ('E' ('+'|'-')? (FLOAT_FRAGMENT | [0-9]+))? ('D' | 'F')?;
 
+
 // Rule #--- <CHAR_STRING> is a base for Rule #065 <char_string_lit> , it incorporates <character_representation>
 // and a superfluous subtoken typecasting of the "QUOTE"
 CHAR_STRING: '\''  (~('\'' | '\r' | '\n') | '\'' '\'' | NEWLINE)* '\'';
 
+Q_STR_BLOCK: 'Q' CHAR_STRING ;
```

Comment:
Please refer to only fragment rules inside other tokens if possible (the only case where it's impossible is recursive rules such as comments). Otherwise, it makes lexer more compicated.

You can rewrite this rule in the following way:

```suggestion
Q_STR_BLOCK: 'Q' CHAR_STRING ;
CHAR_STRING: CHAR_STRING_FRAGMENT;
fragment CHAR_STRING_FRAGMENT: '\''  (~('\'' | '\r' | '\n') | '\'' '\'' | NEWLINE)* '\'';
```